packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -324,25 +325,26 @@ class CreateCommand extends FlutterCommand with CreateBase {
         argResults!.wasParsed('platforms') &&
         platforms.contains('web')) {
       throwToolExit('The web platform is not supported in plugin_ffi template.', exitCode: 2);
-    } else if (generateFfi && argResults!.wasParsed('ios-language')) {
-      throwToolExit(
-        'The "ios-language" option is not supported with the ${template.cliName} '
-        'template: the language will always be C or C++.',
-        exitCode: 2,
-      );
     } else if (generateFfi && argResults!.wasParsed('android-language')) {
       throwToolExit(
         'The "android-language" option is not supported with the ${template.cliName} '
         'template: the language will always be C or C++.',
         exitCode: 2,
       );
     } else if (argResults!.wasParsed('ios-language')) {
-      globals.printWarning(
-        'The "ios-language" option is deprecated and will be removed in a future Flutter release.',
-      );
-      if (stringArg('ios-language') == 'objc') {
+      if (generateMethodChannelsPlugin) {
         globals.printWarning(
-          'Please comment in https://github.com/flutter/flutter/issues/148586 describing your use-case for using Objective-C instead of Swift.',
+          'The "ios-language" option is deprecated and will be removed in a future Flutter release.',
+        );
+        if (stringArg('ios-language') == 'objc') {
+          globals.printWarning(
+            'Please comment in https://github.com/flutter/flutter/issues/148586 describing your use-case for using Objective-C instead of Swift.',
```

Comment:
This will need an update per https://github.com/flutter/flutter/issues/148586#issuecomment-2920134120